src/parser/insert-values-parser.ts

```diff
@@ -91,34 +91,39 @@ function parseRowValues(
 ): PrimitiveValueListNode | ValueListNode {
   const rowColumns = Object.keys(row)
 
-  const rowValues: ValueExpression<any, any, unknown>[] = Array.from({
+  const rowValues = Array.from({
     length: columns.size,
   })
 
-  let complexColumn = false
+  let hasUndefinedOrComplexColumns = false
 
   for (const col of rowColumns) {
     const columnIdx = columns.get(col)
 
-    if (columnIdx !== undefined) {
-      const value = row[col]
+    if (isUndefined(columnIdx)) {
+      continue
+    }
 
-      if (isExpressionOrFactory(value)) {
-        complexColumn = true
-      }
+    const value = row[col]
 
-      rowValues[columnIdx] = value
+    if (isUndefined(value) || isExpressionOrFactory(value)) {
+      hasUndefinedOrComplexColumns = true
     }
+
+    rowValues[columnIdx] = value
   }
 
-  const columnMissing = rowColumns.length < columns.size
+  const hasMissingColumns = rowColumns.length < columns.size
 
-  if (columnMissing || complexColumn) {
-    const defaultValue = DefaultInsertValueNode.create()
+  if (hasMissingColumns || hasUndefinedOrComplexColumns) {
+    let defaultValueNode: DefaultInsertValueNode
 
     return ValueListNode.create(
-      rowValues.map((it) =>
-        isUndefined(it) ? defaultValue : parseValueExpression(it)
+      rowValues.map((value) =>
+        isUndefined(value)
+          ? (defaultValueNode =
```

Comment:
Move the assignment away from the ternary. Do we really need this optimization? I don't think the old way caused any performance problems. We could even create a new instance each time and ditch the `defaultValueNode` variable.

Reply:
I'll revert to the old way.